src/javascript/binary/pages/trade/message.js

```diff
@@ -3,35 +3,82 @@
  * events based on type of response
  */
 var Message = (function () {
-    'use strict';
+    "use strict";
 
     var process = function (msg) {
-        var response = JSON.parse(msg.data);
-        if (response) {
-            var type = response.msg_type;
-            if (type === 'authorize') {
-                TUser.set(response.authorize);
-                TradeSocket.send({ payout_currencies: 1 });
-            } else if (type === 'active_symbols') {
+
+        var response;
+
+        try {
+            response  = JSON.parse(msg.data);
+            if("object" !== typeof response || !("msg_type" in response)) {
+                throw new Error("Response from WS API is not well formatted.");
+            }
+        } catch(e) {
+            throw new Error("Response from WS API is not well formatted.");
+        }
+
+        var msg_type = response.msg_type;
+        
+        switch(msg_type) {
+
+            case "error":
+                $(".error-msg").text(response.error.message);
+                break;
+
+            case "authorize":
+
+                var thendo = "unknown";
```

Comment:
This is the same use-case explained in response to the same question on my other PR here:
https://github.com/binary-com/binary-static/pull/597#discussion_r43570631